packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -410,9 +414,17 @@ export class CronjobController extends BaseController<
    * @returns An array of background events.
    */
   getBackgroundEvents(snapId: SnapId): BackgroundEvent[] {
-    return Object.values(this.state.events).filter(
-      (snapEvent) => snapEvent.snapId === snapId,
-    );
+    return Object.values(this.state.events)
+      .filter((snapEvent) => snapEvent.snapId === snapId)
+      .map((event) => ({
+        ...event,
+        // Truncate dates to remove milliseconds.
+        date: DateTime.fromISO(event.date, { setZone: true })
+          .startOf('second')
+          .toISO({
+            suppressMilliseconds: true,
+          }) as string,
```

Comment:
Maybe move this to a separate function for readability, and so we can add a test for it?

Reply:
Done